pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -197,7 +201,7 @@ def broadcast(self, obj: object, src: int = 0) -> object:
         buffer = io.BytesIO()
         torch.save(obj, buffer)
         data = bytearray(buffer.getbuffer())
-        data_tensor = torch.tensor(data).to(xm.xla_device(), dtype=torch.float)
+        data_tensor = torch.tensor(data).to(self.device, dtype=torch.float)
```

Comment:
This avoids moving from cpu to the device

```suggestion
        data_tensor = torch.tensor(data, device=self.device, dtype=torch.float)
```